app/mailers/api_entreprise/authorization_request_mailer.rb

```diff
@@ -0,0 +1,34 @@
+class APIEntreprise::AuthorizationRequestMailer < APIEntrepriseMailer
+  include ExternalUrlHelper
+
+  %w[
+    enquete_satisfaction
+    embarquement_brouillon_en_attente
+    embarquement_demande_refusee
+    embarquement_modifications_demandees
+    embarquement_relance_modifications_demandees
+    embarquement_valide_to_editeur
+    embarquement_valide_to_demandeur_tech_metier
+    embarquement_valide_to_demandeur_seulement
+    embarquement_valide_to_metier_cc_demandeur_tech
+    embarquement_valide_to_demandeur_not_tech
+    embarquement_valide_to_demandeur_tech_not_metier
+    embarquement_valide_to_tech_cc_demandeur_metier
+    embarquement_valide_to_tech_cc_metier
+    demande_recue
+    reassurance_demande_recue
+  ].each do |method|
+    send('define_method', method) do |args|
+      @all_scopes = I18n.t('api_entreprise.tokens.token.scope')
+      @authorization_request = args[:authorization_request]
+      @authorization_request_scopes = @authorization_request.token.scopes.map(&:to_sym)
+      @authorization_request_datapass_url = datapass_authorization_request_url(@authorization_request)
+
+      @full_name_demandeur = @authorization_request.demandeur.full_name
+      @full_name_contact_technique = @authorization_request.contact_technique&.full_name
+      @full_name_contact_metier = @authorization_request.contact_metier&.full_name
+
+      mail(to: args[:to], cc: args[:cc], subject: I18n.t('api_entreprise.authorization_request_mailer.subject')[__method__]) { |format| format.html }
```

Comment:
```suggestion
      mail(to: args[:to], cc: args[:cc], subject: t('.subject')) { |format| format.html }
```

ne fonctionne pas ? Faut changer dans l'i18n pour rendre ça rails way ofc

Reply:
```I18n.t('api_entreprise.authorization_request_mailer.subject')[__method__]```

Aucune idée de ce que ca, ca fait.

Reply:
`I18n.t('api_entreprise.authorization_request_mailer.subject')` renvoi un hash, `__method__` le nom de la méthode courante

Reply:
ah oui ok on va pas chercher une clé de trad dans i18n mais tout le bloc.

Wah, vous faites des trucs avec les trads qui sont quand même pas banal.

Perso j'aurais fait un fichier yaml à part pour le hash même si c'est pas complètement DRY, là c'est pas vraiment du i18n non plus.

Enfin c'est pas grave :D

Reply:
J'ai proposé quelque chose de plus conventionnel pour le coup

Reply:
J'ai viré la première partie comme suggéré dans ta suggestion mais je vois pas comment virer le [__method__] et que ça marche encore, c'était peut-être un accident dans ta suggestion?

Reply:
`t('.subject')` te renvoi quoi ? parce que imo ça fonctionne si le path est `api_entreprise.authorization_request_mailer.METHOD_NAME.subject`

Reply:
Ah okay! Je fais comme ça.